sync.go

```diff
@@ -255,36 +317,40 @@ func (lw *LibWallet) RescanBlocks() error {
 		progress := make(chan wallet.RescanProgress, 1)
 		ctx, _ := contextWithShutdownCancel(context.Background())
 
-		var totalHeight int32
+		var totalHeightRescanned int32
 		go lw.wallet.RescanProgressFromHeight(ctx, netBackend, 0, progress)
 
 		for p := range progress {
 			if p.Err != nil {
 				log.Error(p.Err)
-
 				return
 			}
-			totalHeight += p.ScannedThrough
+			totalHeightRescanned += p.ScannedThrough
 			for _, syncProgressListener := range lw.syncProgressListeners {
-				syncProgressListener.OnRescan(p.ScannedThrough, SyncStateProgress)
+				report := &HeadersRescanProgressReport{
+					CurrentRescanHeight: p.ScannedThrough,
```

Comment:
`CurrentRescanHeight` should be `totalHeightRescanned`, `p.ScannedThrough` represents the number of headers scanned in this loop, `CurrentRescanHeight` should be the total number of headers scanned to this point.